src/app/profiles/profiles.component.ts

```diff
@@ -0,0 +1,134 @@
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+// 
+// Runbox 7 is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+import { timeout } from 'rxjs/operators';
+import { SecurityContext, Component, Input, Output, EventEmitter, NgZone, ViewChild, AfterViewInit } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { Router } from '@angular/router';
+import { ProgressService } from '../http/progress.service';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { BrowserModule } from '@angular/platform-browser';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import {
+  MatCardModule,
+  MatCheckboxModule,
+  MatDialogModule,
+  MatInputModule,
+  MatListModule,
+  MatPaginatorModule,
+  MatProgressBarModule,
+  MatProgressSpinnerModule,
+  MatSelectModule,
+  MatSnackBarModule,
+  MatTableModule,
+  MatTabsModule,
+  MatChipsModule,
+  MatDialog,
+  MatPaginator,
+  MatSnackBar,
+} from '@angular/material';
+
+import {MatFormFieldModule} from '@angular/material/form-field'; 
+import {AliasesLister} from '../aliases/aliases.lister';
+import {ProfilesLister} from './profiles.lister';
+import {ProfilesEditorModal} from './profiles.editor.modal';
+import {AliasesEditorModal} from '../aliases/aliases.editor.modal';
+import { RMM } from '../rmm';
+
+@Component({
+  moduleId: 'angular2/app/profiles/',
+  selector: 'profiles',
+  templateUrl: 'profiles.component.html'
+})
+
+export class ProfilesComponent implements AfterViewInit {
+  panelOpenState = false;
+  @ViewChild(MatPaginator, { static: false }) paginator: MatPaginator;
+  @Output() onClose: EventEmitter<string> = new EventEmitter();
+  domain;
+  aliases = [];
+  aliases_counter = {};
+  aliases_unique = [];
+  dialog_ref : any;
+
+  ngAfterViewInit() {
+  }
+
+  ev_reload_emiter (ev) {
+    this.load_aliases();
+    this.load_profiles();
+  }
+
+  constructor(
+    public snackBar: MatSnackBar,
+    public dialog: MatDialog,
+    public rmm: RMM,
+  ) {
+    this.rmm.runbox_domain.load();
+    this.load_profiles();
+    this.load_aliases();
+  }
+
+  load_aliases () {
+    this.rmm.alias.load();
+  }
+
+  load_profiles () {
+    this.rmm.profile.load();
+  }
+
+  show_error (message, action) {
+    this.snackBar.open(message, action, {
+      duration: 2000,
+    });
+  };
+
+  add_alias (): void {
+      window.location.href = '/mail/account_alias';
```

Comment:
It seems like either the redirect to rmm6 or the AliasesEditor is unnecessary here – I suspect the former, and in that case the href and return should probably go away.

Reply:
aliases will be added from rmm6